src/Controllers/ElementalAreaController.php

```diff
@@ -195,25 +224,123 @@ public function formAction(HTTPRequest $request)
     }
 
     /**
-     * Remove the pseudo namespaces that were added to form fields by the form factory
-     *
-     * @param array $data
-     * @param int $elementID
-     * @return array
+     * Create a form for a given element
      */
-    public static function removeNamespacesFromFields(array $data, $elementID)
+    private function createElementForm(BaseElement $element): Form
     {
-        $output = [];
-        $template = sprintf(EditFormFactory::FIELD_NAMESPACE_TEMPLATE ?? '', $elementID, '');
-        foreach ($data as $key => $value) {
-            // Only look at fields that match the namespace template
-            if (substr($key ?? '', 0, strlen($template ?? '')) !== $template) {
+        $id = $element->ID;
+        $scaffolder = Injector::inst()->get(EditFormFactory::class);
+
+        /** @var Form $form */
+        $form = $scaffolder->getForm(
+            $this,
+            sprintf(static::FORM_NAME_TEMPLATE, $id),
+            ['Record' => $element]
+        );
+
+        $urlSegment = $this->config()->get('url_segment');
+        $form->setFormAction("admin/$urlSegment/elementForm/$id");
+
+        if (!$element->canEdit()) {
+            $form->makeReadonly();
+        }
+
+        $form->addExtraClass('element-editor-editform__form');
+        $form->addExtraClass('bypass-entwine-submission');
+
+        // Add a save button to the form
+        // This button needs to be present for the save() method to be called
+        // It will be hidden with CSS via the btn--hidden class
+        $form->setActions(FieldList::create([
+            FormAction::create('save', 'Save')
+                ->addExtraClass('btn--hidden')
+        ]));
+
+        // Set the form request handler to return a FormSchema response during a POST request
+        // This will override the default FormRequestHandler::getAjaxErrorResponse() which isn't useful
+        $form->setValidationResponseCallback(function (ValidationResult $errors) use ($form, $id) {
+            $schemaId = Controller::join_links(
+                $this->Link('schema'),
+                $this->config()->get('url_segment'),
+                $id
+            );
+            return $this->getSchemaResponse($schemaId, $form, $errors);
+        });
+
+        return $form;
+    }
+
+    /**
+     * Call model validation taking into account the namespaced fields
+     */
+    private function getElementValidationResult(
+        Form $form,
+        array $data,
+        BaseElement $element
+    ): ValidationResult {
+
+        // Remove the namespace prefixes that were added by EditFormFactory
+        $dataWithoutNamespaces = $this->removeNamespacesFromFields($data, $element->ID);
```

Comment:
```suggestion
        $dataWithoutNamespaces = static::removeNamespacesFromFields($data, $element->ID);
```
Static methods should be called statically.

Reply:
Updated